components/org.wso2.carbon.config/src/test/java/org/wso2/carbon/config/configprovider/ConfigProviderImplTest.java

```diff
@@ -139,6 +139,139 @@ public void yamlFileConfigObjectTestCase() throws IOException {
         }
     }
 
+    @Test(description = "This test will test functionality of creating configuration object with different namespace")
+    public void configObjectWithDifferentNamespaceTestCase() {
+        try {
+            ConfigFileReader fileReader = new YAMLBasedConfigFileReader(TestUtils.getResourcePath("conf", "Example" +
+                    ".yaml").get());
+            ConfigProvider configProvider = new ConfigProviderImpl(fileReader, secureVault);
+            TestConfiguration configurations = configProvider.getConfigurationObject("testconfiguration.v2",
+                    TestConfiguration.class);
+
+            //Transport 1
+            Assert.assertEquals(configurations.getTenant(), "tenant");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(0).getName(), "abc");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(0).getPort(), 9090);
+            Assert.assertEquals(configurations.getTransports().getTransport().get(0).isSecure(), "true");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(0).getDesc(),
+                    "This transport will use 8000 as its port");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(0).getPassword(), PASSWORD);
+
+            //Transport 2
+            Assert.assertEquals(configurations.getTransports().getTransport().get(1).getName(), "pqrs");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(1).getPort(), 8501);
+            Assert.assertEquals(configurations.getTransports().getTransport().get(1).isSecure(), "true");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(1).getDesc(),
+                    "This transport will use 8501 as its port. Secure - true");
+            //Transport 3
+            Assert.assertEquals(configurations.getTransports().getTransport().get(2).getName(), "xyz");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(2).getPort(), 9000);
+            Assert.assertEquals(configurations.getTransports().getTransport().get(2).isSecure(), "true");
+            Assert.assertEquals(configurations.getTransports().getTransport().get(2).getDesc(),
+                    "This transport will use 8888 as its port");
+        } catch (ConfigurationException e) {
+            logger.error(e.toString());
+            Assert.fail();
```

Comment:
Is it required to have a try-catch here. If an exception is thrown it means the test is failed. Therefore shouldn't we have a throws statement in the method signature rather than a try-catch.